homeassistant/components/climate/vera.py

```diff
@@ -0,0 +1,135 @@
+        VeraThermostat(device, VERA_CONTROLLER) for
+        device in VERA_DEVICES['thermostat'])
+
+
+# pylint: disable=abstract-method
+class VeraThermostat(VeraDevice, ClimateDevice):
+    """Representation of a Vera Thermostat."""
+
+    def __init__(self, vera_device, controller):
+        """Initialize the Vera device."""
+        VeraDevice.__init__(self, vera_device, controller)
+        self._fan_list = ["On", "Auto", "Cycle"]
+        self._operation_list = ["Heat", "Cool", "Auto Changeover", "Off"]
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        mode = self.vera_device.get_hvac_mode()
+        if mode == "HeatOn":
+            return self._operation_list[0]  # heat
+        elif mode == "CoolOn":
+            return self._operation_list[1]  # cool
+        elif mode == "AutoChangeOver":
+            return self._operation_list[2]  # auto
+        elif mode == "Off":
+            return self._operation_list[3]  # off
+        return "Off"
+
+    @property
+    def operation_list(self):
+        """List of available operation modes."""
+        return self._operation_list
+
+    @property
+    def current_fan_mode(self):
+        """Return the fan setting."""
+        mode = self.vera_device.get_fan_mode()
+        if mode == "ContinuousOn":
+            return self._fan_list[0]  # on
+        elif mode == "Auto":
+            return self._fan_list[1]  # auto
+        elif mode == "PeriodicOn":
+            return self._fan_list[2]  # cycle
+        return "Auto"
+
+    @property
+    def fan_list(self):
+        """List of available fan modes."""
+        return self._fan_list
+
+    def set_fan_mode(self, mode):
+        """Set new target temperature."""
+        if mode == self._fan_list[0]:
+            self.vera_device.fan_on()
+        elif mode == self._fan_list[1]:
+            self.vera_device.fan_auto()
+        elif mode == self._fan_list[2]:
+            return self.vera_device.fan_cycle()
+
+    @property
+    def current_power_mwh(self):
+        """Current power usage in mWh."""
+        power = self.vera_device.power
+        if power:
+            return convert(power, float, 0.0) * 1000
+
+    def update(self):
+        """Called by the vera device callback to update state."""
+        self._state = self.vera_device.get_hvac_mode()
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return TEMP_FAHRENHEIT
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self.vera_device.get_current_temperature()
+
+    @property
+    def operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        return self.vera_device.get_hvac_state()
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self.vera_device.get_current_goal_temperature()
```

Comment:
Does these methods make a call to the vera device? In that case they should be stored as instance variables inside the `update` method.